modules/integration/src/test/scala/scala/cli/integration/RunScriptTestDefinitions.scala

```diff
@@ -326,4 +326,35 @@ trait RunScriptTestDefinitions { _: RunTestDefinitions =>
         expect(p.out.trim() == "List(1, 2, 3, -v)")
       }
     }
+
+  test("script file with shebang header and no extension run with scala-cli shebang") {
+    val inputs = TestInputs(
+      os.rel / "script-with-shebang" ->
+        s"""|#!/usr/bin/env -S ${TestUtil.cli.mkString(" ")} shebang -S 2.13
+            |//> using scala "$actualScalaVersion"
+            |println(args.toList)""".stripMargin
+    )
+    inputs.fromRoot { root =>
+      os.perms.set(root / "script-with-shebang", os.PermSet.fromString("rwx------"))
+      val output = os.proc("./script-with-shebang", "1", "2", "3", "-v").call(cwd = root).out.trim()
```

Comment:
Alternatively, a version that should work for Windows:
```suggestion
      val output = 
        (if (!Properties.isWin) {
          os.perms.set(root / "script-with-shebang", os.PermSet.fromString("rwx------"))
          os.proc("./script-with-shebang", "1", "2", "3", "-v")
        } else 
          os.proc(TestUtil.cli, "shebang", "script-no-shebang", "1", "2", "3", "-v")
          ).call(cwd = root).out.trim()
```